hibernate-core/src/main/java/org/hibernate/mapping/Value.java

```diff
@@ -179,4 +181,31 @@ default MetadataBuildingContext getBuildingContext() {
 	default String getExtraCreateTableInfo() {
 		return "";
 	}
+
+	/**
+	 * Checks if this value contains any duplicate column. A column
+	 * is considered duplicate when its {@link Column#getName() name} is
+	 * already contained in the {@code distinctColumn} set.
+	 * <p>
+	 * If a duplicate column is found, a {@link MappingException} is thrown.
+	 *
+	 * @param distinctColumns set containing the names of the columns to check
+	 * @param owner the owner of this value, used just for error reporting
+	 */
+	@Internal
+	default void checkColumnDuplication(Set<String> distinctColumns, String owner) {
+		for ( int i = 0; i < getSelectables().size(); i++ ) {
+			final Selectable selectable = getSelectables().get( i );
+			if ( isColumnInsertable( i ) || isColumnUpdateable( i ) ) {
```

Comment:
I actually intentionally omitted it because I thought a formula is *always* non-insertable and non-updatable. Maybe it's a wrong assumption, but that's what it seemed to me looking at `SimpleValue#addFormula`.

Reply:
Alright, that makes sense. Nothing to change then! :)